wallet-api/src/Wallet/Emulator/Client.hs

```diff
@@ -1,30 +1,57 @@
-{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE TypeSynonymInstances #-}
 
 module Wallet.Emulator.Client where
 
+import           Control.Monad.Reader  (ReaderT, asks, lift, runReaderT)
 import           Data.Proxy            (Proxy (Proxy))
 import           Data.Set              (Set)
 import           Servant.API           ((:<|>) ((:<|>)), NoContent)
-import           Servant.Client        (ClientM, client)
+import           Servant.Client        (ClientEnv, ClientM, ServantError, client, runClientM)
+import           Wallet.API            (KeyPair, WalletAPI (..))
 import           Wallet.Emulator.Http  (API)
 import           Wallet.Emulator.Types (Wallet)
 import           Wallet.UTXO           (Block, Height, Tx, TxIn', TxOut', ValidationData, Value)
 
 api :: Proxy API
 api = Proxy
 
-wallets :: ClientM [Wallet]
-fetchWallet :: Wallet -> ClientM Wallet
-createWallet :: Wallet -> ClientM NoContent
-createPaymentWithChange :: Wallet -> Value -> ClientM (Set TxIn', TxOut')
-payToPublicKey :: Wallet -> Value -> ClientM TxOut'
-submitTxn :: Wallet -> Tx -> ClientM ()
-getTransactions :: ClientM [Tx]
-blockchainActions :: ClientM [Tx]
-setValidationData :: ValidationData -> ClientM ()
-blockValidated :: Wallet -> Block -> ClientM ()
-blockHeight :: Wallet -> Height -> ClientM ()
-assertOwnFundsEq :: Wallet -> Value -> ClientM NoContent
-assertIsValidated :: Tx -> ClientM NoContent
-(wallets :<|> fetchWallet :<|> createWallet :<|> createPaymentWithChange :<|> payToPublicKey :<|> submitTxn :<|> getTransactions) :<|> (blockValidated :<|> blockHeight) :<|> (blockchainActions :<|> setValidationData) :<|> (assertOwnFundsEq :<|> assertIsValidated) =
+wallets' :: ClientM [Wallet]
```

Comment:
Why did you prime all of these?

Reply:
to avoid confusion with functions of the same name in the `WalletAPI` class and presumably later other classes.

Reply:
I think that's easy enough to manage with qualified imports. I think it's unlikely that anyone is going ot import both the endpoints and the `WalletAPI` class simultaneously.

Reply:
well nobody should be using these endpoint functions directly, in fact I should hide them now.

Reply:
ok, I've used `'` only on functions that are currently hidden inside this module as they are used in the `WalletAPI` class.